### ISSUE TITLE
How can I obtain the reordered data order？

### ISSUE THREAD

**Issue by Chaos-woo** [itisnajim/reorderable_staggered_scroll_view#2]
I create cube widget list by ReorderableStaggeredScrollViewGridItem, and use ReorderableStaggeredScrollView.grid to wrap items. Now, I want to know how get reordered cube order? for example, my cubes keys is 1,2,3, when i drag 2 behind to 3, I how to get 1,3,2 ? Hope you can see and answer me, thanks!! Here is my code:

```
List<ReorderableStaggeredScrollViewGridItem> items = List.generate(
      widget.cubes.length,
          (index) => ReorderableStaggeredScrollViewGridItem(
        key: ValueKey(index.toString()),
        mainAxisCellCount: widget.cubes[index].definition.height.length,
        crossAxisCellCount: widget.cubes[index].definition.width.length,
        widget: widget.cubes[index],
      ),
    );

    return Obx(() {
      return ReorderableStaggeredScrollView.grid(
        enable: (widget.dragEnabled ?? RxBool(false)).value,
        padding: EdgeInsets.all(widget.padding),
        scrollDirection: Axis.vertical,
        physics: const BouncingScrollPhysics(),
        crossAxisCount: widget.crossAxisCount,
        isLongPressDraggable: false,
        draggingWidgetOpacity: 0.6,
        children: items,
      );
    });
```

**Comment by Chaos-woo** [comment]
And I use onMove function, it throw exception: 
`type 'DragTargetDetails<ReorderableStaggeredScrollViewListItem>' is not a subtype of type 'DragTargetDetails<ReorderableStaggeredScrollViewGridItem>' in type cast`

**Comment by Chaos-woo** [comment]
I find it's my mistake... After I read source code I fond when `setWillAccept` function will change param `widget.dataList`'s data order(use remove and insert), so I custom data list will follow its changes. But I use Android Studio's Restart button to restart project, dataList order will not be changed in picture. Maybe I use const keyword on I custom class `ReorderableCubeFlowContainer`??.... I feel very strange....
![Screenshot 2023-12-09 031419](https://github.com/itisnajim/reorderable_staggered_scroll_view/assets/24642577/b9dd18b9-a193-4837-88c0-3f787ee133e1)

Here is my all code:
```
class ReorderableCubeFlowContainer extends StatefulWidget {
  final List<RCCubeWidget> cubes;
  final int crossAxisCount;
  final RxBool? dragEnabled;
  final double padding;
  final void Function()? onComplete;

  const ReorderableCubeFlowContainer({
    super.key,
    required this.cubes,
    required this.crossAxisCount,
    this.dragEnabled,
    this.padding = 10.0,
    this.onComplete,
  });

  @override
  State<ReorderableCubeFlowContainer> createState() => _ReorderableCubeFlowContainerState();
}

class _ReorderableCubeFlowContainerState extends RUiKeepAliveState<ReorderableCubeFlowContainer> {
  @override
  Widget build(BuildContext context) {
    super.build(context);

    List<ReorderableStaggeredScrollViewGridItem> items = List.generate(
      widget.cubes.length,
          (index) => ReorderableStaggeredScrollViewGridItem(
        key: ValueKey(index.toString()),
        mainAxisCellCount: widget.cubes[index].definition.height.length,
        crossAxisCellCount: widget.cubes[index].definition.width.length,
        widget: widget.cubes[index],
      ),
    );

    return Obx(() {
      return ReorderableStaggeredScrollView.grid(
        enable: (widget.dragEnabled ?? RxBool(false)).value,
        padding: EdgeInsets.all(widget.padding),
        scrollDirection: Axis.vertical,
        physics: const BouncingScrollPhysics(),
        crossAxisCount: widget.crossAxisCount,
        isLongPressDraggable: false,
        draggingWidgetOpacity: 0.6,
        children: items,
        onDragCompleted: (_) {
          var string = '';
          for(var item in items) {
            string += '${item.key},';
          }
          print('$string');
        },
      );
    });
  }
}
```

**Comment by Chaos-woo** [comment]
I must Stop and Start, then it's OK

**Comment by Chaos-woo** [comment]
Maybe I use AutomaticKeepAliveClientMixin component or GetX package's Obx to wrap it, so Adnroid Studio Restart can not work for me. But I click stop and start button in AS it will normally work. 

If you see this issues, hope you can fix `onMove` callback type cast error in grid mode, thanks.

**Comment by itisnajim** [comment]
`onMove` fixed in the new version `0.0.1+3`

**Comment by Phoenix-TU** [comment]
> I create cube widget list by ReorderableStaggeredScrollViewGridItem, and use ReorderableStaggeredScrollView.grid to wrap items. Now, I want to know how get reordered cube order? for example, my cubes keys is 1,2,3, when i drag 2 behind to 3, I how to get 1,3,2 ? Hope you can see and answer me, thanks!! Here is my code:
> 
> ```
> List<ReorderableStaggeredScrollViewGridItem> items = List.generate(
>       widget.cubes.length,
>           (index) => ReorderableStaggeredScrollViewGridItem(
>         key: ValueKey(index.toString()),
>         mainAxisCellCount: widget.cubes[index].definition.height.length,
>         crossAxisCellCount: widget.cubes[index].definition.width.length,
>         widget: widget.cubes[index],
>       ),
>     );
> 
>     return Obx(() {
>       return ReorderableStaggeredScrollView.grid(
>         enable: (widget.dragEnabled ?? RxBool(false)).value,
>         padding: EdgeInsets.all(widget.padding),
>         scrollDirection: Axis.vertical,
>         physics: const BouncingScrollPhysics(),
>         crossAxisCount: widget.crossAxisCount,
>         isLongPressDraggable: false,
>         draggingWidgetOpacity: 0.6,
>         children: items,
>       );
>     });
> ```

I did a pr adding to `onWillAccept` and `onAccept` the list of the keys of the widgets, after changing the order. Can you try it and tell me if it does solve your problem?
